internal/util/parser.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"regexp"
+	"strings"
+)
+
+// This function takes a file path and a pattern that mathes the name of the target function.
+// Returns the comment attached to the target function.
+// Errors occur when the file or target function cannot be found or if the target function does not have a comment attached.
```

Comment:
I'm not convinced about the error semantics here.
Returning an error on no comment (especially when we can return an empty comment (on whitespace only)) seems pretty harsh.
I think that we either need to return an empty string if we find the function (but not the comment),
or return a special error that the caller can check for to see this case
(like [io.EOF](https://pkg.go.dev/io#pkg-variables)).

Handling this can also make your tests for this function a little smoother.

(Note that this is a hard design decision, so there is no clear answer.)